librz/core/cmd/cmd_flirt.c

```diff
@@ -41,3 +41,73 @@ RZ_IPI RzCmdStatus rz_flirt_scan_handler(RzCore *core, int argc, const char **ar
 	rz_cons_printf("Found %d FLIRT signatures via %s\n", new - old, argv[1]);
 	return RZ_CMD_STATUS_OK;
 }
+
+RZ_IPI RzCmdStatus rz_flirt_function_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisFunction *func = rz_analysis_get_function_at(core->analysis, core->offset);
+	if (!func) {
+		RZ_LOG_ERROR("core: Cannot find function at 0x%08" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	ut64 size = rz_analysis_function_size_from_entry(func);
+
+	ut8 *pattern = malloc(size);
+	if (!pattern) {
+		RZ_LOG_ERROR("core: cannot allocate function buffer.\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (!rz_io_read_at(core->io, func->addr, pattern, (int)size)) {
+		RZ_LOG_ERROR("core: couldn't read function %s at 0x%" PFMT64x ".\n", func->name, func->addr);
+		free(pattern);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
```

Comment:
Does this assume a linear function? I think in cmd_analysis.c you iterated over the bbs instead of just getting the function linear size.

Reply:
no, i checked, when the function is not linear the size is till the jump. this is the correct one.

Reply:
The function size is from the entry to the lowest jump. if you have a jump to an higher address, then the function is terminated to the jump. I have verified this with the output from IDA.